### PR TITLE
create maven module 'core'

### PR DESCRIPTION
**Pull request by dsgrieve**
1. Moved current source to module 'core'
2. Updated pom files so 'core' builds and tests. 
3. Did a publishing dry run.

### CODE REVIEW

**Review by jeanbisutti — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -78,6 +79,10 @@
       </build>
     </profile>
   </profiles>
+
+  <modules>
+    <module>core</module>
+  </modules>
   
   <properties>
     <maven.compiler.source>8</maven.compiler.source>
```
[the remark this answers is not in the document]
Comment (dsgrieve) [reply pom.xml:88]: We want this to build with JDK 1.8. Part of the reason for this library is to give access to JFR functionality in 1.8 which has no public API for JFR.

File: pom.xml [hunk 2]
```diff
@@ -106,7 +112,8 @@
       <version>8.2.0</version>
       <scope>test</scope>
     </dependency>
-  </dependencies>
+   </dependencies>
+  </dependencyManagement>
 
   <build>
     <plugins>
```
Comment (jeanbisutti) [inline pom.xml:119]: I may add a Maven profile to not execute the `maven-gpg-plugin` and `nexus-staging-maven-plugin  plugins outside of a release process (as done [here](https://github.com/quick-perf/quickperf/blob/c25891ffd45977d5e681702972b557e6ad4de170/pom.xml#L198))

**Review by johnoliver — APPROVED** [review]

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeanbisutti) [reply pom.xml:88]: > We want this to build with JDK 1.8. Part of the reason for this library is to give access to JFR functionality in 1.8 which has no public API for JFR.

The point is about a Java 8 compilation from a JDK >= 9.

File: pom.xml [hunk 3]
```diff
@@ -106,7 +112,8 @@
       <version>8.2.0</version>
       <scope>test</scope>
     </dependency>
-  </dependencies>
+   </dependencies>
```
[the remark this answers is not in the document]
Comment (jeanbisutti) [reply pom.xml:115]: `It's useful for local builds (picks up what you have first, not what Maven central has)`

The same behavior seems to happen without `<relativePath>../pom.xml</relativePath>`, see for example "By default, Maven looks for the parent POM first at project’s root, then the local repository, and lastly in the remote repository." from this [page](https://howtodoinjava.com/maven/maven-parent-child-pom-example/#parent-content).

**Pull request merged**